account_statement_base_import/models/account_move.py

```diff
@@ -0,0 +1,419 @@
+                raise ErrorTooManyPartner(msg)
+            res['partner_id'] = partners[0].id
+        return res
+
+    def get_from_name_and_partner_name(self, line):
+        """Match the partner based on the label field of the statement line and
+        the name of the partner. Then, call the generic get_values_for_line
+        method to complete other values. If more than one partner matched,
+        raise the ErrorTooManyPartner error.
+
+        :param dict st_line: read of the concerned account.bank.statement.line
+        :return:
+            A dict of value that can be passed directly to the write method of
+            the statement line or {}
+           {'partner_id': value,
+            'account_id': value,
+
+            ...}
+            """
+        res = {}
+        # The regexp_replace() escapes the name to avoid false positive
+        # example: 'John J. Doe (No 1)' is escaped to 'John J\. Doe \(No 1\)'
+        # See http://stackoverflow.com/a/400316/1504003 for a list of
+        # chars to escape. Postgres is POSIX-ARE, compatible with
+        # POSIX-ERE excepted that '\' must be escaped inside brackets according
+        # to:
+        #  http://www.postgresql.org/docs/9.0/static/functions-matching.html
+        # in chapter 9.7.3.6. Limits and Compatibility
+        sql = r"""
+        SELECT id FROM (
+            SELECT id,
+                regexp_matches(%s,
+                    regexp_replace(name,'([\.\^\$\*\+\?\(\)\[\{\\\|])', %s,
+                        'g'), 'i') AS name_match
+            FROM res_partner)
+            AS res_partner_matcher
+        WHERE name_match IS NOT NULL"""
+        self.env.cr.execute(sql, (line.name, r"\\\1"))
+        result = self.env.cr.fetchall()
+        if result:
+            if len(result) > 1:
+                raise ErrorTooManyPartner(
+                    _('Line named "%s" was matched by more than one '
+                      'partner while looking on partner by name') %
+                    line.name)
+            res['partner_id'] = result[0][0]
+        return res
+
+
+class AccountMoveLine(models.Model):
+    """
+    Add sparse field on the statement line to allow to store all the bank infos
+    that are given by a bank/office. You can then add you own in your module.
+    The idea here is to store all bank/office infos in the
+    additionnal_bank_fields serialized field when importing the file. If many
+    values, add a tab in the bank statement line to store your specific one.
+    Have a look in account_statement_base_import module to see how we've done
+    it.
+    """
+    _inherit = "account.move.line"
+    _order = "already_completed desc, date asc"
+
+    already_completed = fields.Boolean(
+        string="Auto-Completed",
+        default=False,
+        help="When this checkbox is ticked, the auto-completion "
+        "process/button will ignore this line.")
+
+    def _get_line_values_from_rules(self, line, rules):
```

Comment:
Can't we make this method an api.multi ? And thus use `self` as a recordset to replace `line`.